test/UnbrandedProjects/Unbranded-TypeSpec/src/Generated/Internal/BinaryContentHelper.cs

```diff
@@ -0,0 +1,120 @@
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.ClientModel;
+using System.ClientModel.Primitives;
+using System.Collections.Generic;
+using System.Text.Json;
+
+namespace UnbrandedTypeSpec
+{
+    internal static class BinaryContentHelper
```

Comment:
Is this basically a copy of Azure.Core's `RequestContentHelper`?

I wonder if, when we work on these types for https://github.com/Azure/azure-sdk-for-net/issues/28369, it would make sense to create collection types that implement `IPersistableModel<T>` and/or `IJsonModel<T>` to handle these out of the box instead of with extension methods?

I don't think that blocks this implementation today, though, but wanted to flag it as future work.

Reply:
Yes.
We have removed this file in shared code, therefore it is always generated by the generator, we could be able to remove this file from shared file, and do not need to migrate it in the future.

This file is to serve the case that we are using types other than models (including arrays, dictionaries, other primitive types), such as operations defined in this way in tsp:
```tsp
op Foo(@body body: Foo[]): void;
```
Because they are built-in types in C#, we cannot let them to implement `IPersistableModel<T>` or `IJsonModel<T>`, also from the name of the interface, they are not "models" therefore it might not make sense if we could.

Our principal is that if someone could write an operation using tsp and the tsp could compile, we should generate proper code for it without violating our patterns of generating them and it should work just fine. The helper is enabling cases such as this.
We could optimize the implementation in the future, I agree with that.